### ISSUE TITLE
Enable/disable schema update per configuration 

### ISSUE THREAD

**Issue by vargbalazs** [typeorm/typeorm#10425]
### Feature Description

As of today, if we have our entity and the data table was already created and we start our NestJs app, TypeORM queries for the schema of the entity. I assume, this is because if we make a change in the entity, TypeORM will update the db accordingly. Which is - I think - completely fine and convenient, if we are in development mode. But if we have many entities, it can cause a certain overhead in the application startup time. And this we don't need in production time.

It would be great, if we could enable/disable this schema query/update per configuration.

### The Solution

TypeORM could introduce a new configuration option, f. e. 'updateSchema' with a boolean value.

If set to true, the db will be queried for schema and any changes in the entity will lead to a db update. If set to false, this wouldn't happen.

Basically, this option would be similar to the 'synchronize' option.

### Considered Alternatives

I don't know any other alternatives.

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by vargbalazs** [comment]
In the meantime, I found out, that this is because by default all entities will be checked for schema changes. We can change this by setting synchronize: false on the entity itself. But if this options exists and is by default true, then it totally overrides the synchronize option on the datasource itself. Is this a bug?
